third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -89,6 +89,7 @@ impl Interpreter {
         ty_args: Vec<Type>,
         args: Vec<Value>,
         data_store: &mut TransactionDataCache,
+        module_store: &ModuleStorageAdapter,
```

Comment:
Shall we use `&dyn ModuleStorage` here for more extensibility in the future?

Reply:
`ModuleStorageAdapter` is just a simple wrapper on `Arc<dyn ModuleStorage>` so what's the difference there? How would using `&dyn ModuleStorage` make it more extensible?

Reply:
I guess a better way to rephrase this is that `&ModuleStorageAdapter`, or effectively `&Arc<dyn ModuleStorage>` is just unnecessarily strict for what's needed for this function. 

`&Arc<dyn ModuleStorage>` requires the storage interface to be passed in to be stored in an `Arc`, which prevents other kinds of storages (e.g. an owned in-memory database) from being passed in.